v2/azure-arm.yaml

```diff
@@ -542,6 +542,23 @@ status:
 #     implemented as an extension on the resource type in question.
 #     Only valid for resource types.
 #
+# $azureGeneratedConfigs: <map of string -> string>
+#     The key of the map is the name of the property to export, the value of the map is a json-path like expression
+#     to the property to export. Collections (or really any fancy jsonpath expressions) are not currently supported.
+#     Only valid for resources.
+#     Example:
+#         $azureGeneratedConfigs:
+#           BlobEndpoint: .Status.PrimaryEndpoints.Blob
```

Comment:
Is the leading period necessary? It'll be really easy to leave out, and I guess it wouldn't work.

Reply:
So my intent here was to mimic JSONPath. I missed that that required a leading `$`. I've changed these to be `$.Status.PrimaryEndpoints.Blob` as you're right, the leading `.` wasn't correct.

You'll get an error if you get the right structure but the wrong names, the error looks like:
```
F0411 15:19:19.345605 4144497 root.go:26] failed during pipeline stage 45/73 [addOperatorSpec]: Adds the property 'OperatorSpec' to all Spec types that require it: finding properties allowed to export as config maps: $azureGeneratedConfigs property "FullyQualifiedDomainName" not found at path "$.Status.FullyQualifiedDomainNamez"
```

I also just added an error for if you forget the `$.`

```
F0411 15:24:23.442393 4144829 root.go:26] configuration file loaded from "azure-arm.yaml" is not valid YAML: decoding yaml for "dbforpostgresql": decoding yaml for "2022-01-20-preview": decoding yaml for "FlexibleServer": $generatedConfigs entry "Status.FullyQualifiedDomainName" must begin with $.
```